apollo-compiler/src/main/kotlin/com/apollographql/apollo/compiler/ir/IRBuilder.kt

```diff
@@ -188,6 +195,21 @@ class IRBuilder(private val schema: IntrospectionSchema,
     return usedTypes
   }
 
+  private fun List<FragmentRef>.referencedRootFragmentNames(operationType: String, fragments: List<Fragment>, filePath: String): Set<String> {
```

Comment:
Mostly nitpicking at this point but it feels a bit "off" that we need a special case for the root fragmentsRefs compared to `field.fragmentRefs`. What about moving the `schema.rootTypeForOperationType(operationType)` to the call site and re-using `List<FragmentRef>.findFragments` ?

Reply:
Ah nice catch. A lot of code here ended up being mostly duplicated from `List<FragmentRef>.findFragments` and I didn't even realize. Will refactor! I think I'll keep this function here though since we still need to do some extra lookups for all the nested fragments inside the root fragment.